netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -182,7 +183,13 @@ public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata.Headers he
     Http2Headers http2Headers = Utils.convertClientHeaders(headers, ssl, defaultPath, authority);
 
     // Write the command requesting the creation of the stream.
-    channel.writeAndFlush(new CreateStreamCommand(http2Headers, stream));
+    CreateStreamCommand msg = new CreateStreamCommand(http2Headers, stream);
+    if (method.getType().clientSendsOneMessage()) {
+      // No need to flush yet if the client is required to send at least one message
+      channel.write(msg);
```

Comment:
@ejona86 so what exactly are you suggesting? Do we need `writeAndFlush` in the `else` case or not?

My general concern with needless flushing (i.e. flushing when there is no outbound message for "this" stream) is that since we're multiplexing multiple streams on the same channel, the extra flushes may cause us to actually write small chunks of data from other streams.

> If the CreateStreamCommand is queued, then it won't need to be marked for future flushing. That's because when it is written it will be written due to some other frame, and that other frame will already need proper flushing.

The other frames may also be buffered by `BufferingHttp2ConnectionEncoder`, so allowing the flush event to propagate through the pipeline immediately may not be correct for them either.  I suppose the `BufferingHttp2ConnectionEncoder` could be made to cache the flush events so that they replay after it writes the last frame it received prior to the flush.